monitor-function-calls/src/agent.spec.js

```diff
@@ -0,0 +1,621 @@
+          }
+        }
+      } else {
+        throw new Error(`Unsupported variable type ${typeof (value)} for comparison`);
+      }
+      overrideArgument(argValues, argNames, argOverride);
+
+      const mockFunctionData = iface.encodeFunctionData(
+        testConfig.functionInConfig.name,
+        argValues,
+      );
+
+      // update mock trace object with encoded function data and correct contract address
+      mockTrace[0].action.input = mockFunctionData;
+      mockTrace[0].action.to = validContractAddress;
+      mockTrace[0].action.from = validContractAddress;
+
+      // update mock transaction event with new mock trace and correct contract address
+      mockTxEvent.traces = mockTrace;
+      mockTxEvent.transaction.to = validContractAddress;
+
+      // run the handler
+      const findings = await handleTransaction(mockTxEvent);
+
+      // create the expected finding
+
+      const argumentData = {};
+      argNames.forEach((name, argIndex) => {
+        if (argumentData[name] == null) {
+          argumentData[name] = argValues[argIndex].toString();
+        }
+      });
+
+      const description = `The ${testConfig.functionInConfig.name} function was invoked in the ${contractName} contract, condition met: ${expression}`;
+
+      const testFindings = [Finding.fromObject({
+        alertId: `${config.developerAbbreviation}-${config.protocolAbbreviation}-FUNCTION-CALL`,
+        description,
+        name: `${config.protocolName} Function Call`,
+        protocol: config.protocolName,
+        severity: FindingSeverity[testConfig.findingSeverity],
+        type: FindingType[testConfig.findingType],
+        metadata: {
+          contractAddress: validContractAddress,
+          contractName,
+          functionName: testConfig.functionInConfig.name,
+          ...argumentData,
+        },
+      })];
+
+      expect(findings).toStrictEqual(testFindings);
+    });
+
+    it('returns no finding if a target contract invokes a monitored function when the expression condition is not met', async () => {
+      let expressionObject;
+      initializeData.contracts.forEach((contract) => {
+        const { functionSignatures } = contract;
+        functionSignatures.forEach((functionSignature) => {
+          if (functionSignature.functionName === testConfig.functionInConfig.name) {
+            expressionObject = functionSignature.expressionObject;
+          }
+        });
+      });
+      const { variableName, operator, value } = expressionObject;
+
+      // encode function data
+      // valid function name with valid arguments
+      const { argValues, argNames } = createMockArgs(testConfig.functionInConfig);
+
+      // override the argument value that corresponds to the expression condition
+      // explicitly set the value so that the condition is not met
+      const argOverride = { name: variableName };
+      if (BigNumber.isBigNumber(value)) {
```

Comment:
Same comment as above about refactoring this logic into a separate helper function